GitCommands/Git/GitModule.cs

```diff
@@ -2984,27 +2985,44 @@ public IReadOnlyList<IGitRef> ParseRefs([NotNull] string refList)
             //
             // Lines may also use \t as a column delimiter, such as output of "ls-remote --heads origin".
 
-            var regex = new Regex(@"^(?<objectid>[0-9a-f]{40})[ \t](?<refname>.+)$", RegexOptions.Multiline);
+            var regex = new Regex(@"^(?<objectid>[0-9a-f]{40})[ \t](?<refname>.+?)(?<dereference>\^\{\})?$", RegexOptions.Multiline);
 
             var matches = regex.Matches(refList);
 
             var gitRefs = new List<IGitRef>();
+            var indexByRefName = new Dictionary<string, int>();
             var headByRemote = new Dictionary<string, GitRef>();
 
             foreach (Match match in matches)
             {
                 var refName = match.Groups["refname"].Value;
                 var objectId = ObjectId.Parse(refList, match.Groups["objectid"]);
                 var remoteName = GitRefName.GetRemoteName(refName);
-                var head = new GitRef(this, objectId, refName, remoteName);
+                bool isDeref = match.Groups["dereference"].Success;
+                var gitRef = new GitRef(this, objectId, refName, remoteName);
 
                 if (GitRefName.IsRemoteHead(refName))
                 {
-                    headByRemote[remoteName] = head;
+                    headByRemote[remoteName] = gitRef;
                 }
                 else
                 {
-                    gitRefs.Add(head);
+                    if (indexByRefName.TryGetValue(refName, out var index))
+                    {
+                        if (isDeref)
```

Comment:
add comment?
// For tags, the git-show-ref contains both reference and the derefenced objectid, the latter is what we are interested in